### PR TITLE
feat(tunnel): wire in the tunnel setup and teardown

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

File: cmd/helm/delete.go [hunk 1]
```diff
@@ -18,12 +18,13 @@ deleting them.
...
-	Short:      "Given a release name, delete the release from Kubernetes",
-	Long:       deleteDesc,
-	RunE:       delRelease,
+	Use:               "delete [flags] RELEASE_NAME",
+	Aliases:           []string{"del"},
+	SuggestFor:        []string{"remove", "rm"},
+	Short:             "Given a release name, delete the release from Kubernetes",
+	Long:              deleteDesc,
+	RunE:              delRelease,
+	PersistentPreRunE: setupConnection,
```
Comment (technosophos) [inline cmd/helm/delete.go]: Asking from sheer ignorance... why use `PersistentPreRunE` instead of `PreRunE`?

File: cmd/helm/helm.go [hunk 2]
```diff
@@ -74,12 +71,32 @@ func main() {
 	}
 }
 
-func bootstrap(c *cobra.Command, args []string) {
+func setupConnection(c *cobra.Command, args []string) error {
+	if tillerHost == "" {
+		// Should failure fall back to default host?
```
Comment (technosophos) [inline cmd/helm/helm.go]: I think the current implementation is good. Falling back to the default host might be a confusing thing for the user.

File: cmd/helm/delete.go (hunk 1 above)
Comment (adamreese) [inline cmd/helm/delete.go]: `Prersistent...` means that it applies to any subcommands

**Comment by technosophos** [comment]
Manually tested as follows:

- Tore down and recreated a kubernetes cluster using the scripts/*
- `helm init`
- `helm install` the alpine chart
- `helm list`
- `helm get` and `helm get manifest`
- `kubectl get po` to make sure I saw things as they should be

Then set up a local tiller, and repeated the above with `--host=:44134`.

**Pull request merged**